eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/LongInterval.java

```diff
@@ -0,0 +1,1092 @@
+                    {
+                        batch.add((int) i);
+                        lastUpdated = (int) i;
+                    }
+                    result.add(batch);
+                    innerFrom = lastUpdated + this.step;
+                }
+            }
+            else
+            {
+                while ((lastUpdated + this.step) >= this.to)
+                {
+                    MutableLongList batch = LongLists.mutable.empty();
+                    for (long i = innerFrom; i >= this.to && batch.size() < size; i += this.step)
+                    {
+                        batch.add((int) i);
+                        lastUpdated = (int) i;
+                    }
+                    result.add(batch);
+                    innerFrom = lastUpdated + this.step;
+                }
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public String toString()
+    {
+        return this.makeString("[", ", ", "]");
+    }
+
+    @Override
+    public LongIterator longIterator()
+    {
+        return new LongIntervalIterator();
+    }
+
+    @Override
+    public long getFirst()
+    {
+        return this.from;
+    }
+
+    @Override
+    public long getLast()
+    {
+        return IntervalUtils.valueAtIndex(this.size() - 1, this.from, this.to, this.step);
+    }
+
+    @Override
+    public long get(int index)
+    {
+        this.checkBounds("index", index);
+        return IntervalUtils.valueAtIndex(index, this.from, this.to, this.step);
+    }
+
+    private void checkBounds(String name, int index)
+    {
+        if (index < 0 || index >= this.size())
+        {
+            throw new IndexOutOfBoundsException(name + ": " + index + ' ' + this);
+        }
+    }
+
+    @Override
+    public int indexOf(long value)
+    {
+        return IntervalUtils.indexOf(value, this.from, this.to, this.step);
+    }
+
+    @Override
+    public int lastIndexOf(long value)
+    {
+        return this.indexOf(value);
+    }
+
+    @Override
+    public ImmutableLongList select(LongPredicate predicate)
+    {
+        return LongArrayList.newList(new SelectLongIterable(this, predicate)).toImmutable();
+    }
+
+    @Override
+    public ImmutableLongList reject(LongPredicate predicate)
+    {
+        return LongArrayList.newList(new SelectLongIterable(this, LongPredicates.not(predicate))).toImmutable();
+    }
+
+    @Override
+    public long detectIfNone(LongPredicate predicate, long ifNone)
+    {
+        return new SelectLongIterable(this, predicate).detectIfNone(predicate, ifNone);
+    }
+
+    @Override
+    public <V> ImmutableList<V> collect(LongToObjectFunction<? extends V> function)
+    {
+        return new CollectLongToObjectIterable<V>(this, function).toList().toImmutable();
+    }
+
+    @Override
+    public LazyLongIterable asReversed()
+    {
+        return ReverseLongIterable.adapt(this);
+    }
+
+    @Override
+    public long sum()
+    {
+        long sum = 0L;
```

Comment:
does the formula based implementation not work for longs? overflow?

Reply:
Should work, will fix